internal/peer/node/start.go

```diff
@@ -818,7 +821,18 @@ func serve(args []string) error {
 	}
 
 	if coreConfig.GatewayOptions.Enabled {
-		logger.Info("Starting peer with Gateway enabled")
+		if coreConfig.DiscoveryEnabled {
+			logger.Info("Starting peer with Gateway enabled")
+			gs, err := gatewayservice.CreateGatewayServer(serverEndorser)
+			if err != nil {
+				logger.Panicf("Failed to start Gateway server: %s", err)
+			}
+			gatewayprotos.RegisterGatewayServer(peerServer.Server(), gs)
+			logger.Info("Gateway server activated")
+
+		} else {
+			logger.Error("Discovery service must be enabled for embedded gateway")
```

Comment:
The gateway will depend on discovery in order to work, so this will be an error in as far as we won't be able to start the gateway server without the discovery server.  However, in this initial commit, we are not using discovery so I have removed the if/else condition entirely for now.